tests/Benchmark/RoutingBench.php

```diff
@@ -0,0 +1,30 @@
+<?php
+declare(strict_types=1);
+
+namespace Cake\Tests\Benchmark;
+
+use Cake\Routing\Route\Route;
+
+class RoutingBench
+{
+    protected Route $route;
+
+    public function __construct()
+    {
+    }
+
+    public function setUpCompile()
```

Comment:
This setup hook will be run for each revolution?

Reply:
It might be before the test is run in general. I'll have to check. I was trying to find a way to create an instance without incurring the cost of new + constructor in the test itself.

If it's invoked  before the entire run only, then we won't have a choice but include more in the test.

Reply:
Seems like it happens before all revolutions which means we're still just executing the cached path for the other iterations so the cost is larger than this.

We'll have to test the full new + compile or not bother with including a small test and just test a full collection compile.